src/rpc/service_eth.go

```diff
@@ -207,6 +210,8 @@ func (service *EthService) SendRawTransaction(ctx context.Context, s string) (*c
 	}
 	utils.Logger.Info().Hex("Incoming tx hash", txHash.Bytes()).Hex("Encrypted tx hash", submitTx.Hash().Bytes()).Msg("Transaction sent")
 
+	metrics.MetricsRequestedGasLimit.WithLabelValues(txHash.String()).Observe(float64(tx.Gas()))
+	metrics.MetricsTotalRequestDuration.WithLabelValues(txHash.String()).Observe(float64(time.Since(timeBefore)))
```

Comment:
time here is in nanoseconds, would be great to update it to ms as more readable and clear